### PR TITLE
Extend DbtDocsLocalOperator with static flag

### LINKED ISSUE #746: Extend DbtDocsLocalOperator with --static flag
dbt docs can be hosted in cloud storage as a static page application. The contents of `catalog.json` and `manifest.json` are then directly inserted into the `index.html`. 

A static page app has advantages. For example: 
- Upload and host directly from cloud storage, instead of a server. Making it cheap and easy
- Single page, making it easy to share around. email, slack etc
- Use Cloud Storage provider permissions on the file

Since a few months, DBT provides a flag to do this
`dbt docs generate --static`

I propose to extend this flag to the DbtDocsLocalOperator
Read more about this feature in the dbt discussion [here](https://github.com/dbt-labs/dbt-docs/issues/53)

I'm happy to work on this :)

### PR DESCRIPTION
**Pull request by joppevos**
## Description

Extends the docsOperator to make use of dbt `--static` flag. The static flag is available from dbt 1.7>

I decided to add the flag through `dbt_cmd_flags`. Other options could be a parameter or the `operator_args`.
There is no official documentation page from DBT on the flag [yet](https://github.com/dbt-labs/docs.getdbt.com/issues/4599). When they do add it, I can trim down our documentation and link directly to dbt.

I contribute to learn and am appreciative of any feedback.

## Related Issue(s)
closes #746 

## Checklist

- [x] I have made corresponding changes to the documentation (if required)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by jbandoro — COMMENTED** [review]
This looks great, thanks @joppevos, I'm sure lots of users would really like this option!

Some minor comments below, and we have to wait until #761 is merged which will fix the failing tests and have that fix merged into this branch for the tests to run first.

File: docs/configuration/generating-docs.rst
Comment (joppevos) [reply docs/configuration/generating-docs.rst:92]: Yes let's do that

**Review by jbandoro — APPROVED** [review]
Thanks @joppevos for the contribution!

**Pull request merged**
